transports/dns/CHANGELOG.md

```diff
@@ -5,6 +5,9 @@
 
 [PR 3715]: https://github.com/libp2p/rust-libp2p/pull/3715
 
+- make DnsConfig::custom infallible
+[PR 4464]: https://github.com/libp2p/rust-libp2p/pull/4464
```

Comment:
This is a breaking change so we need this to be in a `0.41.0` section and the version in the `Cargo.toml` needs to be bumped too! :)

See https://github.com/libp2p/rust-libp2p/blob/master/CONTRIBUTING.md#write-changelog-entries-for-user-facing-changes and the other linked docs.